rag/rag-base/src/commonMain/kotlin/ai/koog/rag/base/files/FileSystemProvider.kt

```diff
@@ -4,240 +4,250 @@ import kotlinx.io.Sink
-     * - `Read`: Functionality to read file content, access file content as a source, and determine file size.
-     *
-     * Implementers of this interface are expected to provide implementations for all functionality
-     * associated with serialization, selection, and reading, ensuring a complete read-only perspective
-     * of the file system or equivalent storage mechanisms.
-     *
-     * This interface is useful in scenarios where modification operations are either undesired or disallowed,
-     * offering controlled access to an underlying file or directory structure.
+     * It provides operations for path serialization, structure navigation, and content reading
+     * in a read-only manner without modifying the filesystem.
      */
-    public interface ReadOnly<Path>: Serialization<Path>, Select<Path>, Read<Path>  {}
+    public interface ReadOnly<Path> : Serialization<Path>, Select<Path>, Read<Path> {}
 
     /**
-     * Provides functionalities for creating, moving, writing, and deleting files or directories
-     * while maintaining a serialization context.
-     *
-     * This interface is designed to interact with a file system hierarchy in a structured way,
-     * allowing operations to manage file content and metadata effectively.
+     * Provides operations for creating, moving, writing, and deleting files or directories.
      *
-     * @param Path A type representing a file system location.
+     * This interface focuses on write operations and complements the read operations
+     * provided by other interfaces.
      */
     public interface Write<Path> : Serialization<Path> {
         /**
-         * Creates a new file or directory within a specified parent directory.
+         * Creates a new file or directory inside [parent] with specified [name] and [type].
+         * Parent directories will be created if they don't exist.
          *
-         * @param parent The path of the parent directory where the item will be created.
-         * @param name The name of the new file or directory to be created.
-         * @param type The type of the item to create, either a file or a directory, as defined by [ai.koog.agents.memory.providers.FileMetadata.FileType].
+         * @param parent The parent directory path.
+         * @param name The name of the new file or directory.
+         *        On Windows platforms, reserved names like "CON",
+         *        "PRN", "AUX", "NUL", "COM1"-"COM9", "LPT1"-"LPT9" are not allowed.
+         * @param type The type (file or directory) to create.
+         * @throws FileAlreadyExistsException if a file or directory with [name] already exists in [parent].
+         * @throws InvalidPathException if [name] is invalid (e.g., contains reserved characters).
+         *         Some implementations may throw a more general IllegalArgumentException or IOException instead.
+         * @throws IOException if any other I/O error occurs.
```

Comment:
I'd suggest making it a little bit more abstract:

```
@throws IOException or its inheritor if a file or directory with [name] already exists in [parent], or [name] is invalid (e.g., contains reserved characters), or if any other I/O error occurs.
```

WDYT? My idea is to cover the general idea but let implementations to decide about some details.

Reply:
`@throws IOException or its inheritor ...` can be used in other places as well